go-controller/pkg/testing/testing.go

```diff
@@ -16,26 +17,63 @@ func init() {
 	format.TruncatedDiff = false
 }
 
+// KCmd is a callback spec returning a k8s exec command
+type KCmd func(cmd string, args ...string) kexec.Cmd
+
 // FakeExec is a convenience struct that wraps testing.FakeExec
 type FakeExec struct {
-	fakeexec.FakeExec
+	// Activate this for a loose comparison of executed OVN commands.
+	// We will in such a case ignore order when comparing all executed commands during the run of a test case.
+	// This is important when defining test cases with multiple resources (or multiple resource watchers) of
+	// the same type and not being able to rely on a deterministic order of incomming watch events.
+	looseCompare     bool
+	commandPool      map[string]KCmd
+	expectedCommands []string
+	executedCommnds  []string
+	KFake            fakeexec.FakeExec
 }
 
 // NewFakeExec returns a new FakeExec with a default LookPathFunc
-func NewFakeExec() *FakeExec {
+func NewFakeExec(looseCompare bool) *FakeExec {
```

Comment:
@alexanderConstantinescu actually to make things a bit clearer, could we make this `newFakeExec(looseCompare bool)` and then add two new public methods

1. `NewFakeExec()` --> calls newFakeExec(false)
2. `NewLooseCompareFakeExec()` --> calls newFakeExec(true)

If we have more options to add in the future we should probably do flags or something liek that, but for now this would make it clearer what option various tests are choosing.